src/main/kotlin/org/rust/lang/core/types/infer/TypeInference.kt

```diff
@@ -648,10 +648,22 @@ class RsFnInferenceContext(
         }
     }
 
-    private fun inferAsyncBlockExprType(blockExpr: RsBlockExpr): TyUnknown {
+    private fun inferAsyncBlockExprType(blockExpr: RsBlockExpr, expected: Ty? = null): Ty {
         require(blockExpr.isAsync)
-        blockExpr.block.inferType()
-        return TyUnknown
+        val expectedOutputTy = lookupFutureOutputTy(expected, blockExpr.knownItems)
+        val outputTy = blockExpr.block.inferTypeCoercableTo(expectedOutputTy)
```

Comment:
Is it really should be `inferTypeCoercableTo(expectedOutputTy)`? Or `inferType(expectedOutputTy)` or `inferType(null)` or a something else? I don't know the correct answer; it should be looked up in rustc

Reply:
Hmm, looks like you should just ignore an expectation just b/c you can't provide any valid expectation - async block have `impl Trait` type that can by unified with itself only. Sooo simple `blockExpr.block.inferType()` should be used here